### ISSUE TITLE
Project maintenance

### ISSUE THREAD

**Issue by noemi-mancini** [auth0-community/auth0-drupal#176]
Hi all, this is not an issue for a bug or a feature request, but a simple investigation about the maintenance of this project.

First of all, I want to thank owners and current maintainers for creating and developing this module, that we are using in some projects of our company, and we find it very useful.

We would like to know if the project is currently being maintained or if we can help out in some way.

**Comment by sourcejockey** [comment]
@noemi-mancini I've been asking this for a while now...and it's complete radio silence.

We've started to port it internally, but we're seeing a lot of PHP8 compatibility issues with some of the dependencies...

**Comment by noemi-mancini** [comment]
Hi @sourcejockey, thanks for your reply. 

It would be a real pity if it's not supported anymore.    
At the moment I'm just upgrading the module to ensure D9 compatibility, but it would be nice if there is a way to help maintain the module.

**Comment by sourcejockey** [comment]
Agreed--we rely on it for a couple of our sites, but we're holding off on PHP8 ONLY because of this module...

**Comment by Renrhaf** [comment]
+1

**Comment by duartegarin** [comment]
+1

**Comment by sourcejockey** [comment]
@noemi-mancini Looks like it's toast as of March 2022.

**Comment by noemi-mancini** [comment]
@sourcejockey just read the email.
We're thinking about taking charge of the maintenance. Do you already have any plans?

**Comment by sourcejockey** [comment]
No; we ported ours over to D9.3, but it's not PHP8 compatible (looks like the PHP SDK isn't super PHP8 friendly), and unless the PHP SDK beta makes it to Prod, we're not going to risk it. We didn't have enough clients looking to use it, rightfully so, it seems. Perhaps a chicken and the egg problem.

**Comment by BalintCsuthy** [comment]
https://community.auth0.com/t/community-repo-deprecation-auth0-drupal/69328

https://www.drupal.org/docs/develop/managing-a-drupalorg-theme-module-or-distribution-project/maintainership/taking-over-unsupported-abandoned-projects

I would suggest to do the drupal.org process for taking over abandoned projects. Then move the codebase and the issues back to drupal.org.

**Comment by noemi-mancini** [comment]
Thanks @BalintCsuthy , we're going to take some actions in the next days.

**Comment by sourcejockey** [comment]
@BalintCsuthy @noemi-mancini 
Tried & failed:
https://www.drupal.org/project/auth0/issues/3146695

**Comment by Pasqualle** [comment]
You can take full ownership if the maintainers are unresponsive, as described in 
https://www.drupal.org/docs/develop/managing-a-drupalorg-theme-module-or-distribution-project/maintainership/taking-over-unsupported-abandoned-projects

Create an issue under "Drupal.org project ownership". Read similar issues: 
https://www.drupal.org/project/issues/projectownership

I think the 
deprecation notice: https://community.auth0.com/t/community-repo-deprecation-auth0-drupal/69328 
D9 issue: https://github.com/auth0-community/auth0-drupal/issues/159
d.o issue https://www.drupal.org/project/auth0/issues/3146695
and this issue
is enough evidence that the project is abandoned and you have tried to contact the current maintainers to offer help with  maintenance.

note: Drupal projects can have multiple maintainers, so it should not a problem if multiple users are applying to be maintainers.

**Comment by noemi-mancini** [comment]
Hi all, sorry for the delay. We hadn't the time yet to take any action, sorry!

Thanks to @Pasqualle for all the info, the multiple maintainers option is really interesting. I'm going to open the request in the next days.

**Comment by matthewmessmer** [comment]
Has there been any update on taking over ownership of the module on d.o?

**Comment by AlexanderAllen** [comment]
Really interested in adopting auth0 along with this module.
Auth0 seems like a really solid authentication solution, and auth0 packagist project is up to date with PHP 8.x:

https://packagist.org/packages/auth0/auth0-php#8.0.4

It would be nice to upgrade this module to use the latest auth0-php library, which supports PHP 8.x!

**Comment by AlexanderAllen** [comment]
Since [D.O. issue #3146695](https://www.drupal.org/project/auth0/issues/3146695) is closed as of 13 Oct 2021, I've created a new issue over at Drupal.org to help take over maintenance of the Drupal module:

- [D.O. issue #3257085](https://www.drupal.org/project/auth0/issues/3257085)

I mentioned @noemi-mancini, @shadcn, and @sourcejockey as potentially interested in becoming maintainers as well!

**Comment by matthewmessmer** [comment]
I've updated the module to work with v7 of the Auth0 SDK which is PHP 8.x compatible. It might still contain some bugs, but once someone is able to get ownership of the project on D.O I can open an issue there for review.

**Comment by rob3000** [comment]
Hey @AlexanderAllen I've just given you access to the project page on D.O as well as others mentioned in https://www.drupal.org/project/auth0/issues/3257085

**Comment by Renrhaf** [comment]
I also updated recently a private fork of the module for that purpose. I'll check/update the PRs when the module will be transfered and a PR opened.

**Comment by AlexanderAllen** [comment]
Hey @rob3000, I just saw this. Thank you so much! I'll verify and let you know if there's any issues.

@matthewmessmer you have my attention! Let's open an issue on the D.O. project for this. Definitely need a branch in there for integrating and testing the latest version of the SDK.

@Renrhaf let's check over at the D.O. issue queue, would very much like your insight as well!

**Comment by AlexanderAllen** [comment]
@noemi-mancini, @matthewmessmer, @Renrhaf, @noemi-mancini @shadcn, @sourcejockey: Now that the D.O. community has access to the module (thanks to @rob3000), I've created [D.O. issue #3258312](https://www.drupal.org/project/auth0/issues/3258312) for discovery and putting the Drupal module back into ship shape with the latest versions of the Auth0 library over at Packagist!

Let's get this party started! Thanks everyone for your contributions to this discussion.

**Comment by sourcejockey** [comment]
This is the best news I've heard all year!!!

**Comment by noemi-mancini** [comment]
Hi all! I see great news here!  
I'm so sorry for not responding in recent weeks, work hit hard. :sweat_smile: 

I'm checking in on D.O. but I think another "noemi" user has been added to maintainers instead of me. My account is this one: https://www.drupal.org/user/472120

**Comment by noemi-mancini** [comment]
Hi all! I see great news here!  
I'm so sorry for not responding in recent weeks, work hit hard. :sweat_smile: 

I'm checking in on D.O. but I think another "noemi" user has been added to maintainers instead of me. My account is this one: https://www.drupal.org/user/472120

**Comment by AlexanderAllen** [comment]
The issue tracker was disabled on the project, so that's probably why the issue links where not working properly.
Now it's working again on D.O. after re-enabling the issue tracker.
We should be tracking and coordinating issues over there instead of Github per community best practices for modules.

**Comment by noemi-mancini** [comment]
Yeah I'm in! Thank you!

I agree, we should use the D.O. issue tracker.

**Comment by sourcejockey** [comment]
Agreed on the D.O. issue tracker. I'm @greatmatter (https://www.drupal.org/u/greatmatter) on D.O., I'm happy to pitch in, though all my mods have been on the old (non PHP8) version of the SDK.

**Comment by AlexanderAllen** [comment]
Hey @noemi-mancini, @sourcejockey, jump in here when you get a chance: https://www.drupal.org/project/auth0/issues/3258312#comment-14367184

@sourcejockey: 
1. I know from the Github history and comments that you've worked on this for a bit, you're in there now :)
2.  Those are some of the questions I ponder on issue #3258312. Input greatly appreciated.

For example:
- I'm 100% fine with using the "old" (5.x) library in the interim if it works. It works because I tested it, but of course our mileage may vary in the near future as 5.x is so old.
- Noemi's fork is also using the 5.x library, initially I'm planning to do a feature branch using this code and library version, just to get something working that can be downloaded from D.O.
- I have a new 9.x-1.x branch (per DO naming conventions), we could for example use a 9.x-1.x with the old library, and a 9.x-2.x with newer versions of the library. But I must be delirious at this point, so again, let's coalesce on a near future plan that everyone's comfortable with!

**Comment by AlexanderAllen** [comment]
@rob3000 do you know what is the deprecation timeline for the auth0 library over at Packagist? I'd be nice to have an idea of when Auth0 5.x is gonna keel over so we can plan ahead and inject some urgency into the community efforts if needed!

Cheers.
